src/ripple/server/impl/BaseWSPeer.h

```diff
@@ -258,6 +264,33 @@ close()
                     std::placeholders::_1)));
 }
 
+template<class Handler, class Impl>
+void
+BaseWSPeer<Handler, Impl>::
+close(const std::uint16_t& code, const std::string& reason)
+{
+    if(! strand_.running_in_this_thread())
+        return post(strand_, std::bind(
+            // disambiguate to this close overload (also above)
+            static_cast<void(BaseWSPeer::*)(const std::uint16_t&, const std::string&)>(&BaseWSPeer::close),
+            impl().shared_from_this(),
+            code, reason));
+    do_close_ = true;
+    if(wq_.empty()){
+      boost::beast::websocket::close_reason _reason;
+      _reason.code = code;
+      _reason.reason = reason;
+      impl().ws_.async_close(
+          _reason,
+          bind_executor(
+              strand_,
+              std::bind(
+                  &BaseWSPeer::on_close,
+                  impl().shared_from_this(),
+                  std::placeholders::_1)));
+    }
+}
+
```

Comment:
This function and the one above are almost identical. The only difference is one passes an empty reason while the other passes the caller given reason. How about having the `close()` method create an empty reason and call the `close(boost::beast::websocket::close_reason const&)` function? (Note: I suggest replacing the `close(std::uint16_t const&, std::string const&)` function with one taking a single `reason` parameter elsewhere in this review).